src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -269,6 +269,19 @@ bool QgsOracleProvider::exec( QSqlQuery &qry, QString sql, const QVariantList &a
       qry.addBindValue( arg );
     }
     res = qry.exec();
+    if ( !res && retry )
+    {
+      // If the connection has been closed because of timeout
+      // it will be great to restart it
```

Comment:
Make connections great again!

More seriously, I'd simply write "Try again once (or N times if you are ok with the proposal below) in case of timeout "